sources/xfscli/src/bin/xfs_admin.rs

```diff
@@ -0,0 +1,211 @@
+use std::ops::Add;
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
+/// change parameters of an XFS filesystem
+#[derive(FromArgs)]
+struct Args {
+    /// enables unwritten extent support on a filesystem that is disabled
+    #[argh(switch, short = 'e')]
+    extflg: bool,
+    /// specifies that the filesystem image to be processed is stored in a regular file at device
+    #[argh(switch, short = 'f')]
+    fs_image: bool,
+    /// enables version 2 log format
+    #[argh(switch, short = 'j')]
+    log2: bool,
+    /// prints the current filesystem label
+    #[argh(switch, short = 'l')]
+    fs_label: bool,
+    /// enable 32bit project identifier support
+    #[argh(switch, short = 'p')]
+    projid32bit: bool,
+    /// prints the current filesystem UUID
+    #[argh(switch, short = 'u')]
+    fs_uuid: bool,
+    /// enable (1) or disable (0) lazy-counters in the filesystem
+    #[argh(option, short = 'c')]
+    lazy_counters: Option<usize>,
+    /// set the filesystem label to label
+    #[argh(option, short = 'L')]
+    label: Option<String>,
+    /// add or remove features on an existing V5 filesystem
+    #[argh(option, short = 'O')]
+    v5_feature: Option<String>,
+    /// set the UUID of the filesystem to uuid
+    #[argh(option, short = 'U')]
+    uuid: Option<String>,
+    /// specifies the device special file where the filesystem's realtime section resides
+    #[argh(option, short = 'r')]
+    rtdev: Option<String>,
+    /// block-device
+    #[argh(positional)]
+    device: Option<String>,
+    /// filesystem's external log location
+    #[argh(option)]
+    logdev: Option<String>,
+    /// prints the version number and exits.
+    #[argh(switch, short = 'V')]
+    version: bool,
+}
+
+static FINDMNT: &str = "findmnt";
+static XFS_DB: &str = "/usr/sbin/xfs_db";
+static XFS_IO: &str = "/usr/sbin/xfs_io";
+static XFS_REPAIR: &str = "/usr/sbin/xfs_repair";
+
+fn main() {
+    let args: Args = argh::from_env();
+
+    // Return the version for arg version as true
+    if args.version {
+        let xfs_db_result = Command::new(XFS_DB)
+            .args(["-p", "xfs_admin", "-V"])
+            .output()
+            .expect("Couldn't run xfs db for version info");
+        print!("{}", String::from_utf8(xfs_db_result.stdout).unwrap());
+        exit(xfs_db_result.status.code().unwrap());
+    }
+
+    let mut db_opts: Vec<&str> = vec![];
+    let mut io_opts: Vec<&str> = vec![];
+    let mut repair_opts: Vec<&str> = vec![];
+
+    let mut lazycount_arg = "lazycount=".to_string();
```

Comment:
Can you add comments on what they are intended to be used for? I think I understand because I've read the other shell script but a reader might not know immediately what they are for and why there are so many being created at once.